Remotion/Data/DomainObjects/Queries/EagerFetching/FetchedRelationDataRegistrationAgentBase.cs

```diff
@@ -70,10 +70,10 @@ private void CheckClassDefinitionOfOriginatingObject (IRelationEndPointDefinitio
       Assertion.DebugAssert(originatingObject.IsNull == false, "originatingObject.IsNull == false");
       Assertion.DebugIsNotNull(originatingObject.ObjectID, "originatingObject.ObjectID != null");
 
-      var relationEndPointDefinitionInheritanceRoot = relationEndPointDefinition.TypeDefinition.GetInheritanceRoot();
-      var originatingObjectInheritanceRoot = originatingObject.ObjectID.ClassDefinition.GetInheritanceRoot();
+      var relationEndPointDefinitionInheritanceRoots = TypeDefinitionHierarchy.GetHierarchyRoots(relationEndPointDefinition.TypeDefinition);
```

Comment:
Just something I remembered from a previous review (only related to this code here because of the question "static or extension"):

When I write tests that exercise extension methods, I try to write the extension method as a static method instead of using the extension syntax. That way, it doesn't look like I'm exercising a method on a mock object:

Extension syntax:
`mock.Object.DoExtensionStuff()`

Static syntax:
`Extensions.DoExtensionStuff(mock.Object)`

Thoughts? I'm thinking this could (should?) be an analyzer warning.